client_api/data_access/gumbo_data_access.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+import pandas as pd
+
+
+class GumboDAO:
+    """Data Access Object for """
+    def __init__(
+        self,
+        psycopg2_connection,
+        sanity_check=True,
+        username=None,
+        autocommit=True
+    ):
+        self.sanity_check = sanity_check
+        self.connection = psycopg2_connection
+        self.connection.autocommit = autocommit
+
+        # set the username for use in audit logs
+        self.username = username or os.getlogin() + " (py)"
+        with self.connection.cursor() as cursor:
+            print("setting username to", self.username)
+            cursor.execute("SET my.username=%s", [self.username])
+
+
+    def get(self, table_name):
+        cursor = self.connection.cursor()
+        select_query = f"select * from {table_name}"
```

Comment:
Oh boy. That is just a little too sql-injection-friendly for my comfort.

Now, I know this service will always be behind authentication and so we _probably_ can trust the input, and so I think it's technically fine ... but it still just makes me nervous on an existential level.

Can we validate that `table_name` is  a valid table before doing this?

(I think it was this way in the original client library ... but in that case we had a SQL connection so the caller could do anything they wanted anyway. This feels different to me as it is run on an external machine and the caller may not have unrestricted access to the DB.

I feel like I'm being pedantic, but I also feel like avoiding SQL injection is a good habit to encourage.)